docs/HelloWorldWalkthrough.md

```diff
@@ -0,0 +1,313 @@
+    ```cpp
+    #include <VbsEnclave\Enclave\Implementations.h>
+    
+    uint32_t VbsEnclave::VTL1_Declarations::DoSecretMath(_In_  std::uint32_t val1, _In_  std::uint32_t val2)
+    {
+    	return val1 * val2;
+    }
+    ```
+1.	Choose Build again. There should be no errors.
+
+### Build the host app
+1. In MyHostApp.vcxproj, add reference to the nuget packages, via Right-Click, Manage NuGet packages and refer to the local nuget feed created earlier:
+   * Microsoft.Windows.VBSEnclave.SDK 
+   * Microsoft.Windows.VBSEnclave.CodeGenerator 
+1. Add the projection layer properties to the project file.
+   * Unload the project file and edit it, adding the following properties. The trust layer is optional for the host, host is assumed if missing.
+   * TIP: To keep these properties in sync between projects, consider moving the first 3 properties into a separate '.props' file and import that into both projects.
+        ```html
+        <!-- ************ -->
+        <!-- VBS Enclave codegen properties for host -->
+        <!-- ************ -->
+        <PropertyGroup>
+           <VbsEnclaveEdlPath>$(ProjectDir)..\MySecretEnclave.edl</VbsEnclaveEdlPath>
+           <VbsEnclaveVtl0ClassName>MySecretEnclave</VbsEnclaveVtl0ClassName>
+           <VbsEnclaveNamespace>VbsEnclave</VbsEnclaveNamespace>
+           <VbsEnclaveVirtualTrustLayer>HostApp</VbsEnclaveVirtualTrustLayer>
+        </PropertyGroup>
+        ```
+1. Reload the project, and then in project properties, set compiler and linker flags as follows:
+   * Compiler -> C/C++ Language Standard, choose **ISO C++20 Standard (/std:c++20)**
+   * Linker –> Input ->Additional Dependencies, edit the list to add "**onecore.lib**"
+1.	Right-Click on the project and choose “Build”, it should succeed. 
+1.	Choose “Show all files” in solution explorer and you should see the ‘Generated Files\VbsEnclave\HostApp” folder. To see the generated code, **Stubs.h** is of most interest to the host app.
+1.	In your “main” method, initialize the enclave and call its methods
+    * Add the host side include file from the SDK nuget package and the new Stubs.h code generated header.
+        ```cpp
+        #include <conio.h>
+        #include <iostream>
+        #include <veil\host\enclave_api.vtl0.h>
+        #include <VbsEnclave\HostApp\Stubs.h>
+        ```
+    * Initialize the enclave and call its interface:
+        ```cpp
+        // Create app+user enclave identity
+        auto ownerId = veil::vtl0::appmodel::owner_id();
+    
+        // Load enclave
+        // We don't want ENCLAVE_VBS_FLAG_DEBUG for a retail build!
+        constexpr int EnclaveCreate_Flags {
+        #ifdef _DEBUG
+            ENCLAVE_VBS_FLAG_DEBUG
+        #endif
+        };
+    
+        #ifndef _DEBUG
+            static_assert(*flags & ENCLAVE_VBS_FLAG_DEBUG) == true, "Do not use DEBUG flag for retail builds");
```

Comment:
>*flags [](http://example.com/codeflow?start=26&length=6)

this still won't compile.  flags doesn't exist until line 289 (3 lines down from here).  and why the dereference?